mathesar_ui/src/components/cell/utils.ts

```diff
@@ -70,9 +72,28 @@ export function getDbTypeBasedInputCap(
   return DataTypes[cellInfo?.type ?? 'string'].getInput(column, config);
 }
 
-export function getColumnIconProps(column: CellColumnLike): IconProps {
+export function getColumnIconProps(column: ProcessedColumn): IconProps {
+  const constraints = [
+    ...column.exclusiveConstraints,
+    ...column.sharedConstraints,
+  ];
+  const hasFKConstraints = constraints.find(
+    (constraint) => constraint.type === 'foreignkey',
+  );
+  const hasPKConstraint = constraints.find(
+    (constraint) => constraint.type === 'primary',
+  );
+
+  if (hasFKConstraints) {
+    return iconTableLink;
+  }
+
+  if (hasPKConstraint) {
+    return iconConstraint;
+  }
+
```

Comment:
As mentioned on the previous comment, we shouldn't be passing down `ProcessedColumn` to the `Cell` and it's related components and utils. Please refer `getCellCap` in the same file which gets `linksToTable` as an optional argument to determine if there is a link present.

It is better to accept an optional argument in the method, for eg,. `linkType?: 'inward' | 'outward'` (or you could come up with a better term), and use that argument to determine the icon.